modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,443 @@
+//
+//  * Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+//  * Neither the names of the copyright holders nor the names of the contributors
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+//
+//This software is provided by the copyright holders and contributors "as is" and
+//any express or implied warranties, including, but not limited to, the implied
+//warranties of merchantability and fitness for a particular purpose are disclaimed.
+//In no event shall copyright holders or contributors be liable for any direct,
+//indirect, incidental, special, exemplary, or consequential damages
+//(including, but not limited to, procurement of substitute goods or services;
+//loss of use, data, or profits; or business interruption) however caused
+//and on any theory of liability, whether in contract, strict liability,
+//or tort (including negligence or otherwise) arising in any way out of
+//the use of this software, even if advised of the possibility of such damage.
+
+/*****************************************************************************************************************\
+*   The interface contains the main descriptors that will be implemented in the descriptor class                  *
+\******************************************************************************************************************/
+
+#include "precomp.hpp"
+#include <stdint.h>
+#ifndef _OPENCV_DESCRIPTOR_HPP_
+#define _OPENCV_DESCRIPTOR_HPP_
+#ifdef __cplusplus
+
+namespace cv
+{
+    namespace stereo
+    {
+        //types of supported kernels
+        enum { Dense_Census, Sparse_Census, StarCensus};
+        enum {ClassicCenterSymetricCensus, ModifiedCenterSymetricCensus};
+        enum {StandardMct,MeanVariation};
+        enum {CensusSingle, MCTSingle};
+        //!Mean Variation is a robust kernel that compares a pixel
+        //!not just with the center but also with the mean of the window
+        struct MVKernel
+        {
+            uint8_t *image[2];
+            int *integralImage[2];
+            MVKernel(uint8_t *img[2], int *integral[2])
+            {
+                image[0] = img[0];
+                image[1] = img[1];
+                integralImage[0] = integral[0];
+                integralImage[1] = integral[1];
+            }
+            void operator()(int rrWidth,int w2, int rWidth, int jj, int j, int c[2]) const
+            {
+                (void)w2;
+                for(int i = 0; i < 2; i++)
+                {
+                    if (image[i][rrWidth + jj] > image[i][rWidth + j])
+                    {
+                        c[i] = c[i] + 1;
+                    }
+                    c[i] = c[i] << 1;
+
+                    if (image[i][rrWidth + jj] > integralImage[i][rWidth + j])
```

Comment:
Shouldn't it be `integralImage[i][rrWidth + jj] > integralImage[i][rWidth + j]`?